django/utils/text.py

```diff
@@ -103,38 +110,46 @@ def handle_endtag(self, tag):
             self.tags.remove(tag)
 
     def handle_data(self, data):
-        if self.func == "words":
-            data = re.split(r"(?<=\S)\s(?=\S)", data)
-            output = escape(" ".join(data[: self.num]))
-        else:
-            output = escape("".join(data[: self.num]))
-
+        data, output = self.process(data)
         data_len = len(data)
-        if self.num < data_len:
-            if self.func == "words" and self.truncate and self.truncate[0] == " ":
-                output = output.strip()
+        if self.remaining < data_len:
+            self.remaining = 0
             self.output += Truncator.add_truncation_text(output, self.truncate)
```

Comment:
Looks like `add_truncation_text()` is a class method, but doesn't reference the class. So it could be a static method, but better hoisted out to global scope to avoid the back reference to `Truncator`?

Reply:
(Added a commit for this.)